### ISSUE TITLE
Output bundle as directory tree for direct browser loading

### ISSUE THREAD

**Issue by psytron** [webpack/webpack#11230]
## Feature request

**What is the expected behavior?**
In addition to creating a bundle file, it would be great to have the option to output directory tree with ES6 modules, ideally retaining the same paths, so modules could be loaded directly into the browser.

**What is motivation or use case for adding/changing the behavior?**
Modern browser enable ES6 modules to be loaded directly into browser and also loaded asynchronously / on demand. It would be great to continue using npm to manage dependencies but to have the option to output the dependencies into a directory tree that can be accessed by web server and loaded directly by the browser, on a per file on demand basis. 

**How should this be implemented in your opinion?**
Perhaps just a configuration switch like static_file_tree=true or browser_direct_es6_static_dir=on

**Are you willing to work on this yourself?**
It seems this kind of thing would require wizard level.

**Comment by sokra** [comment]
Maybe we do that, but I have question: What are the benefit you are expecting from this?

**Comment by psytron** [comment]
Benefits:  
+ Runtime direct editing code in browser dev tools and save to disk
+ Leverage NPM without sacrificing ability for precise control over module loading
+ Debugger breakpointing exactly mirrors source code
+ Promise based import(path).then(obj => <module object>) load optimization

**Comment by diem98** [comment]
does this have any plan？

**Comment by lorenzodallavecchia** [comment]
I also think that this could be useful.

For me, the best part would be using **loaders** without bundling. Webpack has an excellent loader ecosystem that could be leveraged by server-side code and by libraries.

**Comment by alexander-akait** [comment]
bump

**Comment by roninjin10** [comment]
This is something we need for the "faster dev time e.g. no recompile of entire bundle, similar to Snowpack.dev" reason

**Comment by alexander-akait** [comment]
@vankop No, but we need to solve this for https://github.com/webpack/webpack/issues/5866
